gulp/tests/prod-build/prod-build.test.js

```diff
@@ -0,0 +1,235 @@
+		 * as wp-cli will not generate a JSON file.
+		 * See https://github.com/wp-cli/i18n-command/issues/136#issuecomment-453006359
+		 */
+		.filter( function( filePath ) {
+			// Get the file contents
+			const fileContents = fs.readFileSync(
+				filePath,
+				{ encoding: 'utf-8' }
+			);
+
+			return fileContents.includes( '.js' );
+		} );
+
+	let JSONFilePathsArray = [];
+
+	// For each .po file found
+	poFilePathsArray.forEach( ( poFilePath ) => {
+		// Stash the .po file base name
+		const poFileBaseName = path.basename( poFilePath, '.po' );
+		// Get all corresponding .json files
+		const currentJSONFilePathsArray = glob.sync( `${ prodThemePath }/languages/*.json` )
+			/**
+			 * Skip the .json file if it is unrelated
+			 * to the .po file we are checking.
+			 */
+			.filter( function( jsonFilePath ) {
+				// Get the file base name
+				const jsonFileBaseName = path.basename( jsonFilePath, '.json' );
+
+				/**
+				 * The generated JSON files have a hash in the file name
+				 * but always start with the same file base name as the
+				 * .po file they were generated from. So we can check if
+				 * the .json file base name starts with the .po file base name
+				 * to determine which .po file it came from.
+				 */
+				return jsonFileBaseName.startsWith( poFileBaseName );
+			} );
+
+		// Add the current .po files .json files to the main array
+		JSONFilePathsArray = JSONFilePathsArray.concat( currentJSONFilePathsArray );
+	} );
+
+	// Make sure each prod .json file exist.
+	JSONFilePathsArray.forEach( ( filePath ) => {
+		const fileExists = fs.existsSync( filePath );
+		let failMessage = `The expected .json file ${ filePath } does not exist`;
+		expect( fileExists, failMessage ).toBe( true );
+
+		// And that it doesn't have any default strings.
+		const fileContents = fs.readFileSync(
+			filePath,
+			{ encoding: 'utf-8' }
+		);
+		Object.keys( nameFieldDefaults ).forEach( ( key ) => {
+			failMessage = `The file ${ filePath } contains the default string ${ nameFieldDefaults[ key ] }`;
+			expect( fileContents, failMessage ).not.toContain( nameFieldDefaults[ key ] );
+		} );
+	} );
+
+	done();
+} );
+
+test( 'pot file generation', ( done ) => {
+
+	const filePath = paths.languages.potDest;
+	// Test that the .pot file exists
+	const potFileExists = fs.existsSync( filePath );
+	let failMessage = `The expected .pot file ${ filePath } does not exist`;
+	expect( potFileExists, failMessage ).toBe( true );
+
+	const fileContents = fs.readFileSync(
+		filePath,
+		{ encoding: 'utf-8' }
+	);
+
+	// Test that the .pot file contains .php translations
+	failMessage = `The .pot file ${ filePath } does not contain any .php translations`;
+	expect( fileContents, failMessage ).toContain( '.php' );
+	
+	// Test that the .pot file contains .js translations
+	failMessage = `The .pot file ${ filePath } does not contain any .js translations`;
+	expect( fileContents, failMessage ).toContain( '.js' );
```

Comment:
I am testing JS translations in the prod `.pot` file since I am mocking the `editor-filters.js` file from #379